config/routes.rb

```diff
@@ -1,3 +1,8 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+  resources :products, only: [:show]
+  mount Cartify::Engine, at: '/'
+  devise_for :users
+
+  resources :products
```

Comment:
remove this line, you already have the products resource defined on line 2

Reply:
you don't have a products controller?

Reply:
Done

Reply:
(ProductController now replaces MenuController)